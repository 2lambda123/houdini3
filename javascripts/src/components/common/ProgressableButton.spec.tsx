// License: LGPL-3.0-or-later
import * as React from 'react';
import 'jest';
import ProgressableButton from './ProgressableButton'
import toJson from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';

describe('ProgressableButton', () => {
  test('Basic title button works', () => {
    let output = shallow(
      <ProgressableButton onClick={() => console.log('alert!')} title={"nothing"}  data-label="button"/>)
    expect(toJson(output)).toMatchSnapshot()
  })

  test('Progress means we change the title, dont disable and do turn on spinner', () => {
    let output = mount(
      <ProgressableButton onClick={() => console.log('alert!')}
                          title={"nothing"}
                          data-label="button"
                          titleOnProgress={"onProgress"}
                          inProgress={true}
                          disableOnProgress={false}

                          />)
    expect(toJson(output)).toMatchSnapshot()
  })

  test('Progress means we change the title, disable and do turn on spinner', () => {
    let output = mount(
      <ProgressableButton onClick={() => console.log('alert!')}
                          title={"nothing"}
                          data-label="button"
                          titleOnProgress={"onProgress"}
                          inProgress={true}
                          disableOnProgress={true}

      />)
    expect(toJson(output)).toMatchSnapshot()
  })


  test('Disabled manually set overrides whether we disable on progress when in progress', () => {
    let output = mount(
      <ProgressableButton onClick={() => console.log('alert!')}
                          title={"nothing"}
                          data-label="button"
                          titleOnProgress={"onProgress"}
                          inProgress={true}
                          disableOnProgress={false}
                          disabled={true}

      />)
    expect(toJson(output)).toMatchSnapshot()
  })


  test('Disabled manually set overrides whether we disable on progress when NOT in progress', () => {
    let output = mount(
      <ProgressableButton onClick={() => console.log('alert!')}
                          title={"nothing"}
                          data-label="button"
                          titleOnProgress={"onProgress"}
                          inProgress={false}
                          disableOnProgress={true}
                          disabled={true}

      />)
    expect(toJson(output)).toMatchSnapshot()
  })


})