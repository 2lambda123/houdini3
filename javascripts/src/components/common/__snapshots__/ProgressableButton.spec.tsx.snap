// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressableButton Basic title button works 1`] = `
<button
  className="button"
  data-label="button"
  onClick={[Function]}
>
  <span>
    nothing
  </span>
</button>
`;

exports[`ProgressableButton Disabled manually set overrides whether we disable on progress when NOT in progress 1`] = `
<ProgressableButton
  data-label="button"
  disableOnProgress={true}
  disabled={true}
  inProgress={false}
  onClick={[Function]}
  title="nothing"
  titleOnProgress="onProgress"
>
  <button
    className="button"
    data-label="button"
    disabled={true}
    onClick={[Function]}
  >
    <span>
      nothing
    </span>
  </button>
</ProgressableButton>
`;

exports[`ProgressableButton Disabled manually set overrides whether we disable on progress when in progress 1`] = `
<ProgressableButton
  data-label="button"
  disableOnProgress={false}
  disabled={true}
  inProgress={true}
  onClick={[Function]}
  title="nothing"
  titleOnProgress="onProgress"
>
  <button
    className="button"
    data-label="button"
    disabled={true}
    onClick={[Function]}
  >
    <span>
      <i
        className="fa fa-spin fa-spinner"
      />
      onProgress
    </span>
  </button>
</ProgressableButton>
`;

exports[`ProgressableButton Progress means we change the title, disable and do turn on spinner 1`] = `
<ProgressableButton
  data-label="button"
  disableOnProgress={true}
  inProgress={true}
  onClick={[Function]}
  title="nothing"
  titleOnProgress="onProgress"
>
  <button
    className="button"
    data-label="button"
    disabled={true}
    onClick={[Function]}
  >
    <span>
      <i
        className="fa fa-spin fa-spinner"
      />
      onProgress
    </span>
  </button>
</ProgressableButton>
`;

exports[`ProgressableButton Progress means we change the title, dont disable and do turn on spinner 1`] = `
<ProgressableButton
  data-label="button"
  disableOnProgress={false}
  inProgress={true}
  onClick={[Function]}
  title="nothing"
  titleOnProgress="onProgress"
>
  <button
    className="button"
    data-label="button"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      <i
        className="fa fa-spin fa-spinner"
      />
      onProgress
    </span>
  </button>
</ProgressableButton>
`;

exports[`ProgressableButton Title is kept on progress if no titleOnProgress is set 1`] = `
<ProgressableButton
  data-label="button"
  inProgress={true}
  onClick={[Function]}
  title="nothing"
>
  <button
    className="button"
    data-label="button"
    onClick={[Function]}
  >
    <span>
      <i
        className="fa fa-spin fa-spinner"
      />
      nothing
    </span>
  </button>
</ProgressableButton>
`;
